src/tests/ftest/control/dmg_telemetry_io_basic.py

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/python
+"""
+  (C) Copyright 2018-2021 Intel Corporation.
+
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+"""
+from avocado.core.exceptions import TestFail
+
+from ior_test_base import IorTestBase
+from telemetry_test_base import TestWithTelemetry
+
+
+class TestWithTelemetryIOBasic(IorTestBase,TestWithTelemetry):
+    # pylint: disable=too-many-ancestors
+    """Test telemetry engine io basic metrics.
+
+    :avocado: recursive
+    """
+
+    def verify_io_test_metrics(self, io_test_metrics, metrics_data, threshold):
+        """ Verify telemetry io metrics from metrics_data.
+
+        Args:
+            io_test_metrics (list): list of telemetry io metrics.
+            metrics_data (dict): a dictionary of host keys linked to a
+                                 list of io metric names.
+            threshold (int): test io metrics threshold.
+
+        """
+        for host in self.hostlist_servers:
+            self.log.info("==Host: %s", host)
+            for name in io_test_metrics:
+                self.log.info("  --telemetry metric: %s", name)
+                for i, _ in enumerate(metrics_data):
+                    m_data = metrics_data[i]
+                    if i == 0:
+                        self.log.info("   Initial    : %s", m_data[host][name])
+                    else:
+                        self.log.info("   testloop %s: %s", i,
+                                      m_data[host][name])
+                    #Detail for each test io metrics threshold to be updated
+                    self.assertGreaterEqual(m_data[host][name], threshold,
```

Comment:
I'd recommend checking all the values and then reporting each invalid value at the end as opposed to failing the test on the first invalid value.

Reply:
please see the updated.